dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/BaseIdentifiableObject.java

```diff
@@ -802,4 +781,24 @@ public String toString()
             "\"lastUpdated\":\"" + getLastUpdated() + "\" " +
             "}";
     }
+
+    /**
+     * Get Translation value from {@code Set<Translation>} by given locale and
+     * translationKey
+     *
+     * @return Translation value if exists, otherwise return null.
+     */
+    private String getTranslationValue( String locale, String translationKey )
+    {
+        for ( Translation translation : translations )
+        {
+            if ( locale.equals( translation.getLocale() ) && translationKey.equals( translation.getProperty() ) &&
```

Comment:
I always wonder if this should not take care of the locale fallback, for example if you ask for `en_GB` but we only have `en` then looking up `en_GB` should end up using `en` when there is no `en_GB`.